.github/workflows/autofix_linters.yml

```diff
@@ -16,32 +13,39 @@ concurrency:
 jobs:
   autofix:
     runs-on: ubuntu-latest
+    permissions:
+      contents: write
     if: ${{ github.event_name == 'pull_request' && (github.event.label.name == 'autofix') }}
     steps:
+      - name: Calculate number of commits
+        run: echo "PR_FETCH_DEPTH=$(( ${{ github.event.pull_request.commits }} + 1 ))" >> "${GITHUB_ENV}"
+
       - uses: actions/checkout@v4
         with:
-          sparse-checkout: apps
           ref: ${{ github.head_ref }}
+          fetch-depth: ${{ env.PR_FETCH_DEPTH }}
+          sparse-checkout: |
+            packages
+            apps
 
       - uses: pnpm/action-setup@v4
         name: Install pnpm
-        with:
-          run_install: false
 
       - uses: actions/setup-node@v4
         with:
           node-version-file: '.nvmrc'
+          cache: pnpm
 
       - name: Install required dependencies
-        run: pnpm i
+        run: pnpm install --frozen-lockfile
 
       - name: Run Prettier in fix mode
         run: pnpm run format
 
       - name: Commit changes and push to existing branch
         run: |
-          git config --global user.name 'github-tidy-bot'
-          git config --global user.email 'github-tidy-bot@supabase.com'
+          git config --global user.name 'kevcodez'
+          git config --global user.email 'k.grueneberg1994@gmail.com'
```

Comment:
Needs to be some user that has write access to the repo, otherwise builds are not running one committed - we can use another one too